database/schema/ve_db_postgresql.sql

```diff
@@ -404,6 +405,7 @@ GRANT DELETE ON system_vulnerabilities, system_vulnerabilities_active, system_vu
 GRANT DELETE ON system_vulnerabilities, system_vulnerabilities_active, system_vulnerabilities_inactive TO ve_db_user_vmaas_sync;
 
 CREATE INDEX IF NOT EXISTS sv_rule_id ON system_vulnerabilities_active (rule_id);
+CREATE INDEX IF NOT EXISTS not_null_advisories_idx ON system_vulnerabilities_active (advisories) WHERE advisories IS NOT NULL;
```

Comment:
description in Jira said `use index to filter NULL values`, so I didn't really think about it 😄 I'll investigate what's better

Reply:
if we are not going to include unpatched CVEs in system_vulnerabilities table, then the `IS NOT NULL` doesn't make much sense, since every CVE should have some errata.

Do `manually_fixable_cves` have advisories?

Some examples with different indexes (but I guess I need to try it on bigger DB) with `SET enable_seqscan = off`

**No advisories index:**
```
vulnerability=# explain select * from system_vulnerabilities_active where advisories like '%2020%' and rh_account_id = 1;
                                                     QUERY PLAN
---------------------------------------------------------------------------------------------------------------------
 Bitmap Heap Scan on system_vulnerabilities_active_184  (cost=26.66..50.28 rows=327 width=160)
   Recheck Cond: (rh_account_id = 1)
   Filter: (advisories ~~ '%2020%'::text)
   ->  Bitmap Index Scan on system_vulnerabilities_active_184_rh_account_id_idx  (cost=0.00..26.58 rows=841 width=0)
         Index Cond: (rh_account_id = 1)
(5 rows)
```

**advisories index**
```
vulnerability=# create index advisories_idx on system_vulnerabilities_active (advisories);
CREATE INDEX
vulnerability=# explain select * from system_vulnerabilities_active where advisories like '%2020%' and rh_account_id = 1;
                                                     QUERY PLAN
---------------------------------------------------------------------------------------------------------------------
 Bitmap Heap Scan on system_vulnerabilities_active_184  (cost=26.66..50.28 rows=327 width=160)
   Recheck Cond: (rh_account_id = 1)
   Filter: (advisories ~~ '%2020%'::text)
   ->  Bitmap Index Scan on system_vulnerabilities_active_184_rh_account_id_idx  (cost=0.00..26.58 rows=841 width=0)
         Index Cond: (rh_account_id = 1)
(5 rows)
```

**advisories is not null index**
```
vulnerability=# create index advisories_idx on system_vulnerabilities_active (advisories) where advisories is not null;
CREATE INDEX
vulnerability=# explain select * from system_vulnerabilities_active where advisories like '%2020%' and rh_account_id = 1;
                                                     QUERY PLAN
---------------------------------------------------------------------------------------------------------------------
 Bitmap Heap Scan on system_vulnerabilities_active_184  (cost=26.66..50.28 rows=327 width=160)
   Recheck Cond: (rh_account_id = 1)
   Filter: (advisories ~~ '%2020%'::text)
   ->  Bitmap Index Scan on system_vulnerabilities_active_184_rh_account_id_idx  (cost=0.00..26.58 rows=841 width=0)
         Index Cond: (rh_account_id = 1)
(5 rows)
```

**gin trigram index**
```
vulnerability=# create index advisories_idx on system_vulnerabilities_active using gin (advisories gin_trgm_ops);
CREATE INDEX
vulnerability=# explain select * from system_vulnerabilities_active where advisories like '%2020%' and rh_account_id = 1;
                                                    QUERY PLAN
------------------------------------------------------------------------------------------------------------------
 Bitmap Heap Scan on system_vulnerabilities_active_184  (cost=14.54..30.44 rows=327 width=160)
   Recheck Cond: (advisories ~~ '%2020%'::text)
   Filter: (rh_account_id = 1)
   ->  Bitmap Index Scan on system_vulnerabilities_active_184_advisories_idx  (cost=0.00..14.46 rows=327 width=0)
         Index Cond: (advisories ~~ '%2020%'::text)
(5 rows)
```

Only the GIN index was used

Reply:
> if we are not going to include unpatched CVEs in system_vulnerabilities table, then the IS NOT NULL doesn't make much sense, since every CVE should have some errata.
> Do manually_fixable_cves have advisories?

Manually fixable CVEs have advisories, but they are not applicable to the customer environment for some reason (e.g. due to not enabled repo). They are CVEs with advisory detected by OVAL using repository-CPE mappings but their advisory is not found in yum/dnf repositories. It's mostly because single CPE is used to describe multiple similar RHEL repositories. Unfortunately it's causing some false positives due to errors in mappings. `manually_fixable_cves` are basically CVEs where something is off between repositories and OVAL.

TLDR: `manually_fixable_cves` `advisory` column should be `NULL` because VMaaS doesn't return the advisory.